tests/test_ceres.py

```diff
@@ -867,3 +869,394 @@ def test_correct_size_written_one_point_gap(self, open_mock):
+      self.assertEqual([600, 320, 120, 0, 0], slice_timestamps)
+
+  def test_archives_no_data_exists_if_no_slices_exist(self):
+    with patch('ceres.CeresNode.readSlices', new=Mock(return_value=[])):
+      self.assertFalse(self.ceres_node.hasDataForInterval(0, 60))
+
+  def test_archives_no_data_exists_if_no_slices_exist_and_no_time_specified(self):
+    with patch('ceres.CeresNode.readSlices', new=Mock(return_value=[])):
+      self.assertFalse(self.ceres_node.hasDataForInterval(None, None))
+
+  def test_archives_data_exists_if_slices_exist_and_no_time_specified(self):
+    with patch('ceres.CeresNode.slices', new=self.ceres_slices):
+      self.assertTrue(self.ceres_node.hasDataForInterval(None, None))
+
+  def test_archives_data_exists_if_slice_covers_interval_completely(self):
+    with patch('ceres.CeresNode.slices', new=[self.ceres_slices[0]]):
+      self.assertTrue(self.ceres_node.hasDataForInterval(1200, 1800))
+
+  def test_archives_data_exists_if_slice_covers_interval_end(self):
+    with patch('ceres.CeresNode.slices', new=[self.ceres_slices[0]]):
+      self.assertTrue(self.ceres_node.hasDataForInterval(600, 1260))
+
+  def test_archives_data_exists_if_slice_covers_interval_start(self):
+    with patch('ceres.CeresNode.slices', new=[self.ceres_slices[0]]):
+      self.assertTrue(self.ceres_node.hasDataForInterval(1740, 2100))
+
+  def test_archives_no_data_exists_if_slice_touches_interval_end(self):
+    with patch('ceres.CeresNode.slices', new=[self.ceres_slices[0]]):
+      self.assertFalse(self.ceres_node.hasDataForInterval(600, 1200))
+
+  def test_archives_no_data_exists_if_slice_touches_interval_start(self):
+    with patch('ceres.CeresNode.slices', new=[self.ceres_slices[0]]):
+      self.assertFalse(self.ceres_node.hasDataForInterval(1800, 2100))
+
+  def test_archives_compact_returns_empty_if_passed_empty(self):
+    self.assertEqual([], self.ceres_node.compact([]))
+
+  def test_archives_compact_filters_null_values(self):
+    self.assertEqual([], self.ceres_node.compact([(60, None)]))
+
+  def test_archives_compact_rounds_timestamps_down_to_step(self):
+    self.assertEqual([[(600, 0)]], self.ceres_node.compact([(605, 0)]))
+
+  def test_archives_compact_drops_duplicate_timestamps(self):
+    datapoints = [(600, 0), (600, 0)]
+    compacted = self.ceres_node.compact(datapoints)
+    self.assertEqual([[(600, 0.0)]], compacted)
+
+  @skip("XXX: Ceres should keep the last of duplicate points")
+  def test_archives_compact_keeps_last_seen_duplicate_timestamp(self):
+    datapoints = [(600, 0), (600, 1)]
+    compacted = self.ceres_node.compact(datapoints)
+    self.assertEqual([[(600, 1.0)]], compacted)
+
+  def test_archives_compact_groups_contiguous_points(self):
+    datapoints = [(600, 0), (630, 0), (840, 0)]
+    compacted = self.ceres_node.compact(datapoints)
+    self.assertEqual([[(600, 0), (630, 0)], [(840, 0)]], compacted)
+
+  def test_archives_write_noops_if_no_datapoints(self):
```

Comment:
Remove this and other write tests from CeresArchiveNodeTest and CeresArchiveNodeReadTest as they're generally not relevant. The one thing we'd possibly want tests for is something to ensure writes go to the timestep in self.timeStep and not some other series. Given https://github.com/graphite-project/ceres/blob/master/ceres.py#L521-L523 clearly skips those though, I consider that low priority